libraries/chain/controller.cpp

```diff
@@ -5213,7 +5213,7 @@ uint32_t controller::last_irreversible_block_num() const {
 }
 
 block_id_type controller::last_irreversible_block_id() const {
-   return my->fork_db_root_block_id();
+   return my->fork_db_has_root() ?  my->fork_db_root_block_id() : block_id_type{};
```

Comment:
I understand this reasoning. I think a cost/benefit analysis would make me prefer returning the sensible default rather than throwing, but it is a matter of opinion I guess.

Reply:
Should I change it in this PR (so that the root() access APIs throw when root is not set?)

Reply:
Personally I prefer the `assert` (which `shared_ptr.operator->()` already provides) to an exception.

Reply:
Sure, I'm fine with that, but then I'll expose `fork_db.has_root()` so that the tester can check.

Reply:
done!